File: 1798-max-number-of-k-sum-pairs/1798-max-number-of-k-sum-pairs.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        int numP = 0;
        unordered_map<int, int> us;
        for (int i: nums){
            if (us.find(i) == us.end()){
                us[i] = 1;
            } else {
                us[i]++;
            }
        }
        
        for (pair<int, int> c: us){
            if (us.find(k - c.first) == us.end()){
                continue;
            } else {
                numP += min(us[k - c.first], us[c.first]);
            }
        }

        return numP / 2;
    }
};

File: 1798-max-number-of-k-sum-pairs/1798-max-number-of-k-sum-pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        nums.sort()
        print(nums)
        st = 0
        end = len(nums) - 1
        numP = 0
        while st < end:
            if nums[st] + nums[end] < k:
                st = st + 1
            elif nums[st] + nums[end] > k:
                end = end - 1
            else:
                numP += 1
                st += 1
                end -= 1
        return numP
        